crates/noirc_evaluator/src/brillig/brillig_gen/brillig_block.rs

```diff
@@ -444,6 +444,10 @@ impl<'block> BrilligBlock<'block> {
 
                     self.brillig_context.deallocate_register(radix);
                 }
+                Value::Intrinsic(Intrinsic::AssertConstant) => {
+                    // Do nothing here as Brillig should be able to handle
+                    // runtime values everywhere including loop bounds and array indices
+                }
```

Comment:
Was this breaking something before? As a user I wouldn't expect `assert_constant` in a brillig function to do nothing.

Reply:
This was panic'ing for me with Brillig as `slice_insert` calls `assert_constant` internally. I was under the impression that assert constant was included only to be clear to users that we are using a constant where it is required. However, this differs from ACIR to Brillig where in Brillig we can use runtime values. I think now that the Brillig optimizations PR is in though I no longer hit the panic on assert_constant so I have removed this match.